diamm/serializers/website/city.py

```diff
@@ -29,12 +33,31 @@ class Meta:
 class CityDetailSerializer(serializers.HyperlinkedModelSerializer):
     archives = ArchiveCitySerializer(many=True)
     country = CountryCitySerializer(
-        source="parent"
+        source='parent'
     )
+    sources = serializers.SerializerMethodField() 
 
     class Meta:
         model = GeographicArea
-        fields = ('url', 'name', 'archives', 'country')
+        fields = ('url', 'name', 'archives', 'country', 'sources')
         extra_kwargs = {
             'url': {'view_name': 'city-detail'}
         }
+
+    def get_sources(self, obj):
+        sources = list()
+        connection = pysolr.Solr(settings.SOLR['SERVER'])
+        fq = ['type:source', 'geographic_area_ii:' + str(obj.id)]
+        results = connection.search('*:*', fq=fq)
```

Comment:
if you pass the `fl` parameter you can restrict it to just the fields you need to retrieve. I read somewhere that the speed of the request is `O(n)` on the number of fields, so restricting them to only the ones you need is a (small) optimization.